interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php

```diff
@@ -3378,6 +3391,19 @@ public function logCCDA($pid, $encounter, $content, $time, $status, $user_id, $d
 
         $mimeType = "text/xml";
 
+        // Let listeners know we are about to save a C-CDA
+        $send = array("CCDA" => $content, 'pid' => $pid);
+        $event = new ServiceSaveEvent($patientService, $send);
```

Comment:
I recommend using the CDADocumentService in your ServiceSaveEvent.  The PatientService has a number of listeners in it and those listeners will have a problem if it receives a non patient record.

If you use the CDADocumentService and your $send represents a ccda table record entry then just changing the CCDA column to be **ccda_content** should make it so everything plays nice in the event system.